# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../linux26/macbook-touch.patch
# Copyright (C) 2006 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

MacBook C2D ID.

  - Rene Rebe <rene@exactcode.de>

--- linux-2.6.19/drivers/usb/input/appletouch.c.orig	2006-11-29 22:57:37.000000000 +0100
+++ linux-2.6.19/drivers/usb/input/appletouch.c	2006-12-24 16:15:21.383185034 +0100
@@ -47,6 +47,8 @@
 #define GEYSER3_ISO_PRODUCT_ID	0x0218
 #define GEYSER3_JIS_PRODUCT_ID	0x0219
 
+#define GEYSER3_ISO2_PRODUCT_ID	0x021B
+
 #define ATP_DEVICE(prod)					\
 	.match_flags = USB_DEVICE_ID_MATCH_DEVICE |		\
 		       USB_DEVICE_ID_MATCH_INT_CLASS |		\
@@ -72,6 +74,8 @@
 	{ ATP_DEVICE(GEYSER3_ISO_PRODUCT_ID) },
 	{ ATP_DEVICE(GEYSER3_JIS_PRODUCT_ID) },
 
+	{ ATP_DEVICE(GEYSER3_ISO2_PRODUCT_ID) },
+
 	/* Terminating entry */
 	{ }
 };
@@ -174,7 +178,8 @@
 
 	return (productId == GEYSER3_ANSI_PRODUCT_ID) ||
 		(productId == GEYSER3_ISO_PRODUCT_ID) ||
-		(productId == GEYSER3_JIS_PRODUCT_ID);
+		(productId == GEYSER3_JIS_PRODUCT_ID) ||
+		(productId == GEYSER3_ISO2_PRODUCT_ID);
 }
 
 static int atp_calculate_abs(int *xy_sensors, int nb_sensors, int fact,
